From 2548438e326d20eda2eb4398139164f5400d22ac Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Mon, 23 Nov 2020 17:54:34 +0100
Subject: [PATCH] CMake: Change generated resources naming

Align generated resources .qrc file naming to qmake naming.
Update tests.

Fixes: QTBUG-88581
Pick-to: 6.0
Change-Id: Id8a2f91f23c257e5b8bd371748c2151ec54a6418
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
---

diff --git a/src/corelib/Qt6CoreMacros.cmake b/src/corelib/Qt6CoreMacros.cmake
index b7fa724..68c5ac4 100644
--- a/src/corelib/Qt6CoreMacros.cmake
+++ b/src/corelib/Qt6CoreMacros.cmake
@@ -1166,7 +1166,7 @@
         return()
     endif()
     list(APPEND output_targets ${output_target_quick})
-    set(generatedBaseName "generated_${newResourceName}")
+    set(generatedBaseName "${newResourceName}")
     set(generatedResourceFile "${CMAKE_CURRENT_BINARY_DIR}/.rcc/${generatedBaseName}.qrc")
 
     # Generate .qrc file:
diff --git a/tests/auto/cmake/test_add_resources_binary_generated/CMakeLists.txt b/tests/auto/cmake/test_add_resources_binary_generated/CMakeLists.txt
index c783934..a764e68 100644
--- a/tests/auto/cmake/test_add_resources_binary_generated/CMakeLists.txt
+++ b/tests/auto/cmake/test_add_resources_binary_generated/CMakeLists.txt
@@ -18,15 +18,15 @@
 
 target_compile_definitions(test_add_resources_binary_generated
                   PRIVATE
-                      RESOURCE1_FULL_PATH="${CMAKE_CURRENT_BINARY_DIR}/generated_resources1.rcc")
+                      RESOURCE1_FULL_PATH="${CMAKE_CURRENT_BINARY_DIR}/resources1.rcc")
 target_compile_definitions(test_add_resources_binary_generated
                   PRIVATE
                       RESOURCE2_FULL_PATH="${CMAKE_CURRENT_BINARY_DIR}/resources2_custom.rcc")
 
-if(NOT TARGET binary_resource_generated_resources1)
+if(NOT TARGET binary_resource_resources1)
     message(FATAL_ERROR "Generated resources binary target was not created")
 endif()
 
-if(NOT TARGET binary_resource_generated_resources2)
+if(NOT TARGET binary_resource_resources2)
     message(FATAL_ERROR "Generated resources binary target was not created")
 endif()
